target-gen/src/cargo.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use cargo_metadata::Message;
+use xshell::{cmd, Shell};
+
+pub fn generate_elf() -> Result<String> {
```

Comment:
This seems like a lot of work to run cargo and parse all the data. Wouldn't it be easier to just accept the elf file as a parameter?

Then target-gen test could be used as a cargo runner for example.

Reply:
Easier is relative. I want to take all effort from the user. But I agree, a runner would be nice :)

Reply:
I think it's a bit limiting for the user if cargo is called automatically, it's more flexible if it's possible to just point the tool to an ELF file. This would also make it possible to use it with other files, not only with cargo projects.

And for this case, I think a runner works exactly the same, since we just extract the elf file here. But when we do a runner cargo will give us the path to the elf file as a parameter, so it ends up being the same.

Reply:
Fair enough, that makes everything simpler :)